### PR TITLE
switch from yarn to npm

### PR DESCRIPTION
**Pull request by ledhed2222**
Title says it all

### CODE REVIEW

**Review by mvadari — COMMENTED** [review]
Should this be merged into `master` and then your feature branch rebased on top of it?

**Comment by thejohnfreeman** [comment]
What is driving this change? Is NPM faster and deterministic now?

**Comment by ledhed2222** [comment]
@mvadari what do you think about just staging all this stuff in the feature branch and cutting a release after? easier for me to manage that way

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ on:
     branches: [ master ]
   pull_request:
     branches: [ master ]
+  workflow_dispatch:
```
Comment (ledhed2222) [inline .github/workflows/nodejs.yml:11]: allows us to manually run tests, which i think is v helpful.

**Comment by mvadari** [comment]
> @mvadari what do you think about just staging all this stuff in the feature branch and cutting a release after? easier for me to manage that way

Yeah that's reasonable.

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@ on:
     branches: [ master ]
   pull_request:
     branches: [ master ]
```
Comment (mvadari) [inline .github/workflows/nodejs.yml:10]: If you're editing this file anyways...
```suggestion
  pull_request:
```
This way, tests are run even on PRs that are children of other branches, such as branches you're merging into a feature branch.

File: .github/workflows/nodejs.yml [hunk 3]
```diff
@@ -3,11 +3,7 @@
 
 name: Node.js CI
 
-on:
-  push:
-    branches: [ master ]
-  pull_request:
-    branches: [ master ]
+on: [push, pull_request, workflow_dispatch]
```
Comment (mvadari) [inline .github/workflows/nodejs.yml:6]: I think on `push` we still want the branch param, but not for `pull_request` and `workflow_dispatch`. If it's on `push`, each test will run twice for every push to a branch with a PR.

**Review by mvadari — APPROVED** [review]

**Pull request merged**